comp/core/flare/helpers/helpers.go

```diff
@@ -0,0 +1,179 @@
+// is being added to ensure that it contains no credentials or unnecessary user-specific data. The FlareBuilder scrubs
+// secrets that match pre-programmed patterns, but it is always better to not capture data containing secrets, than to
+// scrub that data.
+//
+// Everytime a file is copied to the flare the original permissions and ownership of the file is recorded (Unix only).
+//
+// There are reserved path in the flare: "permissions.log" and "flare-creationg.log" (both at the root of the flare).
+// Note as well that the flare does nothing to prevent files to be overwritten by different calls. It's up to the caller
+// to make sure the path used in the flare doesn't clash with other modules.
+type FlareBuilder interface {
+	// AddFile creates a new file in the flare with the content.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/file"). Any necessary directory will
+	// automatically be created.
+	//
+	// 'content' is automatically scrubbed of any sensitive informations before being added to the flare.
+	AddFile(destFile string, content []byte) error
+
+	// AddFileFromFunc creates a new file in the flare with the content returned by the callback.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/file"). Any necessary directory will
+	// automatically be created.
+	//
+	// If the 'cb' returns an error, the file will not be created, the error is added to the flare's logs and returned to
+	// the caller.
+	//
+	// The data returned by 'cb' is automatically scrubbed of any sensitive informations before being added to the flare.
+	AddFileFromFunc(destFile string, cb func() ([]byte, error)) error
+
+	// WriteFileFromFunc creates a new file in the flare and call cb with an io.Writer to write to that file.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/file"). Any necessary directory will
+	// automatically be created.
+	//
+	// The data written by 'cb' to the io.Writer will be first stored in a buffer before being scrubbed and flush to disk.
+	// This is because the scrubber works on whole line, so partial writes could bypass its detection.
+	//
+	// If the 'cb' returns an error, the file will not be created, the error is added to the flare's logs and returned to
+	// the caller.
+	WriteFileFromFunc(destFile string, cb func(io.Writer) error) error
+
+	// CopyFile copies the content of 'srcFile' to the root of the flare.
+	//
+	// 'destFile' is a path relative to the flare root (ex: "some/path/to/a/file"). Any necessary directory will
+	// automatically be created.
+	//
+	// The data is automatically scrubbed of any sensitive informations before being copied.
+	//
+	// Example: CopyFile("/etc/datadog/datadog.yaml") will create a copy of "/etc/datadog/datadog.yaml", named
+	// "datadog.yaml", at the root of the flare.
+	CopyFile(srcFile string) error
```

Comment:
Those are to match required logic from the "pkg/flare", they provide high level helper. I don't think it matters how many time they are called since they answer a behavior currently in the flare package. Removing them would break retro-compatibility.

Reply:
After discussion: we'll  merge `CopyFileRename` `CopyFileTo` `CopyFileToRename` to a single func (will check the edge case of a file have slash or backslash in it).